packages/flutter_tools/test/integration.shard/hot_reload_test.dart

```diff
@@ -16,65 +17,80 @@ import 'test_utils.dart';
-  FlutterRunTestDriver _flutter;
+  final HotReloadProject project = HotReloadProject();
+  FlutterRunTestDriver flutter;
 
   setUp(() async {
     tempDir = createResolvedTempDirectorySync('hot_reload_test.');
-    await _project.setUpIn(tempDir);
-    _flutter = FlutterRunTestDriver(tempDir);
+    await project.setUpIn(tempDir);
+    flutter = FlutterRunTestDriver(tempDir);
   });
 
   tearDown(() async {
-    await _flutter?.stop();
+    await flutter?.stop();
     tryToDelete(tempDir);
   });
 
   test('hot reload works without error', () async {
-    await _flutter.run();
-    await _flutter.hotReload();
+    await flutter.run();
+    await flutter.hotReload();
   });
 
   test('newly added code executes during hot reload', () async {
     final StringBuffer stdout = StringBuffer();
-    final StreamSubscription<String> subscription = _flutter.stdout.listen(stdout.writeln);
-    await _flutter.run();
-    _project.uncommentHotReloadPrint();
+    final StreamSubscription<String> subscription = flutter.stdout.listen(stdout.writeln);
+    await flutter.run();
+    project.uncommentHotReloadPrint();
     try {
-      await _flutter.hotReload();
+      await flutter.hotReload();
       expect(stdout.toString(), contains('(((((RELOAD WORKED)))))'));
     } finally {
       await subscription.cancel();
     }
   });
 
-  test('reloadMethod triggers hot reload behavior', () async {
+  test('fastReassemble behavior triggers hot reload behavior with evaluation of expression', () async {
     final StringBuffer stdout = StringBuffer();
-    final StreamSubscription<String> subscription = _flutter.stdout.listen(stdout.writeln);
-    await _flutter.run();
-    _project.uncommentHotReloadPrint();
+    final StreamSubscription<String> subscription = flutter.stdout.listen(stdout.writeln);
+    await flutter.run(withDebugger: true);
+
+    final int port = flutter.vmServicePort;
+    final VmService vmService = await vmServiceConnectUri('ws://localhost:$port/ws');
     try {
-      final String libraryId = _project.buildBreakpointUri.toString();
-      await _flutter.reloadMethod(libraryId: libraryId, classId: 'MyApp');
-      // reloadMethod does not wait for the next frame, to allow scheduling a new
-      // update while the previous update was pending.
-      await Future<void>.delayed(const Duration(seconds: 1));
-      expect(stdout.toString(), contains('(((((RELOAD WORKED)))))'));
+      // Since the single-widget reload feature is not yet implemented, manually
+      // evaluate the expression for the reload.
+      final Isolate isolate = await waitForExtension(vmService);
+      final LibraryRef actualRef = isolate.libraries.firstWhere((LibraryRef libraryRef) {
+        return libraryRef.uri == 'package:flutter/src/widgets/binding.dart';
+      });
+      final Response response = await vmService.evaluate(
+        isolate.id,
+        actualRef.id,
+        '((){_debugFastReassembleMethod=(Object _) => _ is MyApp})()',
```

Comment:
The magic is here.

Reply:
_debugFastReassembleMethod might need to become public in the future, in case I can't evaluate in the scope of the binding library and still use whatever type.